### PR TITLE
chore: update storybook to version 7

### LINKED ISSUE #2772: Update to Storybook 7
## Description

We should update to Storybook 7 because it may causes a problem with our testing tool Chromatic. It seems to be a problem with pnpm and Chromatic versions.
Storybook 7.0 will considerably improve working with pnpm.

## Proposal

Update to version 7

### PR DESCRIPTION
**Pull request by sarahgm**
closes #2772

### CODE REVIEW

File: config/storybook/main.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+import type { StorybookConfig } from '@storybook/react-vite';
 import path from 'node:path';
 import { sync as findUpSync } from 'find-up';
-import TsconfigPathsPlugin from 'tsconfig-paths-webpack-plugin';
+import { mergeConfig } from 'vite';
 
 const parent = path.resolve(__dirname, '..');
 const root = path.dirname(findUpSync('package.json', { cwd: parent }) || '.');
-const configFile = findUpSync('tsconfig.json', { cwd: root });
+const viteTsConfigPaths = require('vite-tsconfig-paths').default;
```
Comment (sebald) [inline config/storybook/main.ts:8]: No `import` possible? (move up to imports)

File: config/storybook/preview-head.html [hunk 2]
```diff
@@ -3,3 +3,6 @@
   href="https://fonts.googleapis.com/css2?family=Inter:wght@400;600;700&display=swap"
   rel="stylesheet"
 />
+<script>
+  window.global = window;
```
Comment (sebald) [inline config/storybook/preview-head.html:7]: what is this for? :-O

File: package.json [hunk 3]
```diff
@@ -62,5 +62,11 @@
     "release:notification": "zx scripts/create-slack-notification.mjs",
     "slack": "zx scripts/slack-release-notification.mjs --quiet",
     "prepare": "husky install"
+  },
+  "devDependencies": {
```
Comment (sebald) [inline package.json:66]: wie only have dependencies in the root, if they are in the root can we remove storybook from the `config` too? and remove it from the workspaces.

**Review by sebald — DISMISSED** [review]

File: config/storybook/preview-head.html (hunk 2 above)
Comment (sarahgm) [reply config/storybook/preview-head.html:7]: for Storybook to work correctly, without it throws an error: `Uncaught ReferenceError: global is not defined
    at ../../node_modules/.pnpm/jest-mock@27.5.1/node_modules/jest-mock/build/index.js `

File: config/storybook/main.ts (hunk 1 above)
Comment (sarahgm) [reply config/storybook/main.ts:8]: what do you mean?

**Review by sebald — APPROVED** [review]

**Pull request merged**